### ISSUE TITLE
ZeroPadding in the vailla_encoder

### ISSUE THREAD

**Issue by bobvo23** [divamgupta/image-segmentation-keras#247]
This is a question for my understanding.

In the vanilla_encoder, many ZeroPadding2d((1,1)) layer is introduced. May I know the reason for those paddings? Do they reserve the edge information?

The layer high and width is divisible for 2 without the padding.
Thank you for your help.

`def vanilla_encoder(input_height=224,  input_width=224):

    kernel = 3
    filter_size = 64
    pad = 1
    pool_size = 2

    if IMAGE_ORDERING == 'channels_first':
        img_input = Input(shape=(3, input_height, input_width))
    elif IMAGE_ORDERING == 'channels_last':
        img_input = Input(shape=(input_height, input_width, 3))

    x = img_input
    levels = []
    # from 224 to 226, padding 1 zero at each dimention
    x = (ZeroPadding2D((pad, pad), data_format=IMAGE_ORDERING))(x)
    x = (Conv2D(filter_size, (kernel, kernel),
                data_format=IMAGE_ORDERING, padding='valid'))(x)
    x = (BatchNormalization())(x)
    x = (Activation('relu'))(x)
    x = (MaxPooling2D((pool_size, pool_size), data_format=IMAGE_ORDERING))(x)
    levels.append(x)

    x = (ZeroPadding2D((pad, pad), data_format=IMAGE_ORDERING))(x)
    x = (Conv2D(128, (kernel, kernel), data_format=IMAGE_ORDERING,
                padding='valid'))(x)
    x = (BatchNormalization())(x)
    x = (Activation('relu'))(x)
    x = (MaxPooling2D((pool_size, pool_size), data_format=IMAGE_ORDERING))(x)
    levels.append(x)

    for _ in range(3):
        x = (ZeroPadding2D((pad, pad), data_format=IMAGE_ORDERING))(x)
        x = (Conv2D(256, (kernel, kernel),
                    data_format=IMAGE_ORDERING, padding='valid'))(x)
        x = (BatchNormalization())(x)
        x = (Activation('relu'))(x)
        x = (MaxPooling2D((pool_size, pool_size),
                          data_format=IMAGE_ORDERING))(x)
        levels.append(x)

    return img_input, levels`

**Comment by bobvo23** [comment]
It seems that the "valid" padding option creates issues for the output dimension. Instead of using an additional padding layer, may we just change the valid padding to same?
It seems to work for me. May I create a PR to improve the codebase?
